wd_helper.au3

```diff
@@ -378,9 +383,11 @@ Func _WD_WaitElement($sSession, $sStrategy, $sSelector, $iDelay = Default, $iTim
 			__WD_Sleep(10)
 			$iErr = @error
 		WEnd
+		$_WD_DEBUG = $_WD_DEBUG_Saved ; restore DEBUG level
 	EndIf
 
-	Return SetError(__WD_Error($sFuncName, $iErr), 0, $sElement)
+	Local $sMessage = '   Options=' & $iOptions
```

Comment:
I'm not seeing how displaying the value of $iOptions would help with debugging. Do you think this is beneficial?

Reply:
```
__WD_Post: URL=HTTP://127.0.0.1:4444/session/14d64e0a-31f4-4b9f-8d6a-3ce23f9ba017/element; $sData={"using":"xpath","value":"//body/div[1][@aria-hidden=\"true\"]"}
__WD_Post ==> No match (8) : HTTP status = 404 ResponseText={"value":{"error":"no such element","message":"Unable to locate element: //body/div[1][@aria-hidden=\"true\"]","stacktrace":"WebDriverError@chrome://remote/content/shared/webdriver/Errors.jsm:183:5\nNoSuchElementError@chrome://remote/content/shared/webdriver/Errors.jsm:395:5\nelement.find/</<@chrome://remote/content/marionette/element.js:300:16\n"}}
_WD_FindElement ==> No match (8) :    Strategy=xpath   Selector=//body/div[1][@aria-hidden=\"true\"]
_WD_WaitElement ==> Success (0) :    Options=8
```

in this way you can find out why `_WD_WaitElement ==> Success (0)` in case when `_WD_FindElement ==> No match (8)`

Reply:
Sorry, but I don't understand your statement / logic. How does displaying `Options=8` help with debugging?

Reply:
When you see such log:
```
__WD_Post: URL=HTTP://127.0.0.1:4444/session/14d64e0a-31f4-4b9f-8d6a-3ce23f9ba017/element; $sData={"using":"xpath","value":"//body/div[1][@aria-hidden=\"true\"]"}
__WD_Post ==> No match (8) : HTTP status = 404 ResponseText={"value":{"error":"no such element","message":"Unable to locate element: //body/div[1][@aria-hidden=\"true\"]","stacktrace":"WebDriverError@chrome://remote/content/shared/webdriver/Errors.jsm:183:5\nNoSuchElementError@chrome://remote/content/shared/webdriver/Errors.jsm:395:5\nelement.find/</<@chrome://remote/content/marionette/element.js:300:16\n"}}
_WD_FindElement ==> No match (8) :    Strategy=xpath   Selector=//body/div[1][@aria-hidden=\"true\"]
_WD_WaitElement ==> Success (0) :    Options=8
```

Is it Success or Error ?
Only having informational about Optoins you can answer on this question, or you are able to answer without this information ?

Reply:
Not true IMO. `_WD_WaitElement ==> Success (0)` clearly shows success. Adding `Options=8` does help to clarify the type of "wait" that was performed, so in this case you need to know that "8" means "wait until not found"

Reply:
Ok. You have right.
My question should be modified:

```
Is it Success or Error ?
Only having informational about Optoins you can answer on this question, or you are able to answer without this information ?
```
How do you know that the errror form `_WD_FindElement`  is not important and only informative ?

Reply:
> Adding `Options=8` does help to clarify the type of "wait" that was performed, so in this case you need to know that "8" means "wait until not found"

For me this "8" clearly indicates that the previous error message from `_WD_FindElement` was expected and is correct, so it does not need repair anything, and no further analyzis is required.
In relation to `_WD_FindElement`  returns error, without "8" this log will be a guess.

Reply:
> Is it Success or Error ?

You have both in this situation. _WD_FindElement returns an error while _WD_WaitElement indicates success.

> or you are able to answer without this information ?

> How do you know that the errror form _WD_FindElement is not important and only informative ?

Since we know that _WD_WaitElement internally calls _WD_FindElement, we can infer the use of $_WD_OPTION_NoMatch when _WD_FindElement fails yet _WD_WaitElement succeeds.

> For me this "8" clearly indicates that the previous error message from _WD_FindElement was expected and is correct

This requires you to know that "8" means "no match". I can foresee someone making incorrect assumptions about this because the eights in `No match (8)` and `Options=8` don't correlate.

Reply:
> This requires you to know that "8" means "no match". I can foresee someone making incorrect assumptions about this because the eights in `No match (8)` and `Options=8` don't correlate.

Quite possible. However, it is important that wise users in debugging mode have complementary information.

Reply:
I'm waiting for your next comment.

Reply:
I think this discussion has run its course. We can leave this change in place.

Reply:
Agree.